### PR TITLE
mon/LogMonitor: separate out summary by channel

### PR DESCRIPTION
**Pull request by liewegas**
This is a stopgap to keep N entires per channel instead of N total, preventing one log channel
(e.g., audit) from aging out entries from antoher channel (e.g., cluster).

We should still do a larger restructure/refactor on LogMonitor later.

### CODE REVIEW

**Comment by liewegas** [comment]
@jecluis I did the lame thing since it is quick and easy.  This should avoid the major pain point for mimic while we sort out a proper refactor.

File: src/mon/LogMonitor.cc [hunk 1]
```diff
@@ -428,29 +428,57 @@ bool LogMonitor::preprocess_command(MonOpRequestRef op)
...
     ostringstream ss;
-    for (; rp != summary.tail.rbegin(); --rp) {
-      if (!match(*rp)) {
-        continue;
+    if (channel == "*") {
+      list<LogEntry> full_tail;
+      summary.build_ordered_tail(&full_tail);
+      derr << "full " << full_tail << dendl;
+      auto rp = full_tail.rbegin();
+      for (; num > 0 && rp != full_tail.rend(); ++rp) {
```
Comment (jcsp) [inline src/mon/LogMonitor.cc]: Can we avoid writing out this loop twice?  Maybe pull it up into a lambda at the start of the function and then apply it to `full_tail` or `tail_by_channel.find(channel)` conditionally

File: src/mon/LogMonitor.cc [hunk 2]
```diff
@@ -563,12 +591,7 @@ void LogMonitor::check_sub(Subscription *s)
 
   if (s->next == 0) { 
     /* First timer, heh? */
-    bool ret = _create_sub_summary(mlog, sub_level);
-    if (!ret) {
-      dout(1) << __func__ << " ret = " << ret << dendl;
-      mlog->put();
-      return;
-    }
+    _create_sub_incremental(mlog, sub_level, get_last_committed());
```
Comment (jcsp) [inline src/mon/LogMonitor.cc]: Is this going to work if the latest message `sub_level` is in an older revision than `get_last_committed`?

Comment (jcsp) [reply src/mon/LogMonitor.cc]: Never mind, the old _create_sub_summary was only looking at the current summary anyway

File: src/mon/LogMonitor.cc (hunk 1 above)
Comment (liewegas) [reply src/mon/LogMonitor.cc]: It's awkward because one is a list of entries and the other is a list of pairs.. not sure how to generalize that?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-04-14_03:18:44-rados-wip-kefu-testing-2018-04-14-0918-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-04-15_01:40:50-rados-wip-kefu-testing-2018-04-14-0918-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-04-15_13:37:16-rados-wip-kefu-testing-2018-04-14-2138-distro-basic-mira/

the failure is addressed by #21433

**Pull request merged**